feat(sources/elasticsearch): add Elasticsearch source and tools

Anaethelion opened the pull request:
Add support for Elasticsearch with the following tools:
* search
* esql
* get_mappings
* list_indices

This PR fixes #859

averikitsch commented:
Thank you @Anaethelion for the PR! We are a little behind on reviewing new sources/tools right now. We will get to your contribution shortly. Thanks again.

duwenxin99 commented on internal/sources/elasticsearch/elasticsearch_test.go:
```diff
@@ -0,0 +1,67 @@
+		{
+			desc: "basic example",
+			in: `
+sources:
```
Could you format this like the example [here](https://github.com/googleapis/genai-toolbox/blob/9a55b804821a6ccfcd157bcfaee7e599c4a5cb63/internal/sources/bigquery/bigquery_test.go#L36)

duwenxin99 reviewed (changes requested):
Could you add integration tests following the instruction [here](https://github.com/googleapis/genai-toolbox/blob/main/CONTRIBUTING.md#3-add-integration-tests)? Thank you. The elasticsearch-esql tool should be used for these predefined integration tests. For other tools, you can write your own test function. Thanks!

duwenxin99 commented:
Hi @Anaethelion, just a gentle reminder here to update the PR. please let me know if you have any questions about the implementation.

Anaethelion commented:
@duwenxin99 I've updated the branch to the latest from main and added an initial config for Elasticsearch in the cloudbuild. Let me know if you think this misses anything!

duwenxin99 commented:
/gcbrun

duwenxin99 commented:
Hi @Anaethelion, a gentle bump on this PR. Any question regarding my previous comments I'd be happy to answer.

duwenxin99 commented:
/gcbrun

duwenxin99 commented:
@Anaethelion Could you rebase on the latest main? We disabled the dgraph integration test which was failing.

nayrttam commented:
Hi @duwenxin99 - Is there anything we can help with on this (cc: @Anaethelion)?

duwenxin99 commented:
Hi @Anaethelion @nayrttam, sorry for the late update I was distracted by some other tasks. Just took some time to fix the connection issue and triggered the integration test, which is currently failing. Could you rebase and fix the integration test locally? The guide to running integration test locally is [here](https://github.com/googleapis/genai-toolbox/blob/main/DEVELOPER.md#running-locally). Set the `CLIENT_ID` to `32555940559.apps.googleusercontent.com`. Let me know if you have any questions.

Anaethelion commented:
Hi @duwenxin99, we should be good now, `golangci-lint run` is clean on my end.

duwenxin99 commented:
/gcbrun
